packages/dev/loaders/src/glTF/2.0/Extensions/KHR_node_selectability.ts

```diff
@@ -0,0 +1,152 @@
+            inputBlockIndex: 2,
+            outputBlockIndex: 3,
+            isVariable: true,
+        },
+        {
+            input: "object",
+            output: "pickedMesh",
+            inputBlockIndex: 2,
+            outputBlockIndex: 0,
+            isVariable: true,
+        },
+    ],
+    extraProcessor(gltfBlock, _declaration, _mapping, _arrays, serializedObjects, context, globalGLTF) {
+        // add the glTF to the configuration of the last serialized object
+        const serializedObject = serializedObjects[serializedObjects.length - 1];
+        serializedObject.config = serializedObject.config || {};
+        serializedObject.config.glTF = globalGLTF;
+        // find the listener nodeIndex value
+        const nodeIndex = gltfBlock.configuration?.["nodeIndex"]?.value[0];
+        if (nodeIndex === undefined || typeof nodeIndex !== "number") {
+            throw new Error("nodeIndex not found in configuration");
+        }
+        const variableName = "pickedMesh_" + nodeIndex;
+        // find the nodeIndex value
+        serializedObjects[1].config.variable = variableName;
+        context._userVariables[variableName] = {
+            className: "Mesh",
+            id: globalGLTF?.nodes?.[nodeIndex]._babylonTransformNode?.id,
+            uniqueId: globalGLTF?.nodes?.[nodeIndex]._babylonTransformNode?.uniqueId,
+        };
+        return serializedObjects;
+    },
+});
+
+// object model extension for selectable
+addObjectAccessorToKey("/nodes/{}/extensions/KHR_node_selectability/selectable", {
+    get: (node: INode) => {
+        const tn = node._babylonTransformNode as any;
+        if (tn && tn.isPickable !== undefined) {
+            return tn.isPickable;
+        }
+        return true;
+    },
+    set: (value: boolean, node: INode) => {
+        node._primitiveBabylonMeshes?.forEach((mesh) => {
+            mesh.isPickable = value;
+        });
+    },
+    getTarget: (node: INode) => node._babylonTransformNode,
+    getPropertyName: [() => "isPickable"],
+    type: "boolean",
+});
+
+/**
+ * Loader extension for KHR_selectability
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export class KHR_node_selectability implements IGLTFLoaderExtension {
+    /**
+     * The name of this extension.
+     */
+    public readonly name = NAME;
+    /**
+     * Defines whether this extension is enabled.
+     */
+    public enabled: boolean;
+
+    private _loader: GLTFLoader;
+
+    /**
+     * @internal
+     */
+    constructor(loader: GLTFLoader) {
+        this._loader = loader;
+        this.enabled = loader.isExtensionUsed(NAME);
+    }
+
+    public async onReady(): Promise<void> {
+        this._loader.gltf.nodes?.forEach((node) => {
+            if (node.extensions?.KHR_node_selectability && node.extensions?.KHR_node_selectability.selectable === false) {
+                // TODO - this works on load, but will not work if a pointer changes the value in real time
```

Comment:
Do we need to address this?

Reply:
It is addressed in the set function, i simply didn't remove the todo. I'll remove it and resolve after.